jasmine/spec/inverted-index-test.js

```diff
@@ -1,67 +1,164 @@
-      mapData = index.createIndex(myDoc);
-    });
-
-    it("index should be created after reading", () => {
-
-      expect(populatedData).not.toBe(null);
-
-    });
-    it("ensures a correct index is created", () => {
-      expect(mapData['in']).toEqual([0]);
-      expect(mapData['of']).toEqual([0, 1]);
-      expect(mapData['lord']).toEqual([1]);
-      expect(mapData['alice']).toEqual([0]);
-    });
-
-    describe('searchIndex', () => {
-      let getData;
-      beforeEach(() => {
-        getData = index.searchIndex('wonderland');
-      });
-      it('Searches for a particular word', () => {
-        expect(getData).toEqual({ wonderland: [0] });
-      });
-    });
-
-    describe('Correct Index', () => {
-      let getData;
-      beforeEach(() => {
-        getData = index.searchIndex('alice');
-      });
-      it('Finds the correct index of word input', () => {
-        expect(getData).toEqual({ alice: [0] });
-      });
-    });
-  });
-});
\ No newline at end of file
+ const myDoc = [
+   {
+     "title": "Alice in Wonderland",
+     "text": "Alice falls into a rabbit hole and enters a world full of imagination."
+   },
+   {
+     "title": "The Lord of the Rings: The Fellowship of the Ring.",
+     "text": "An unusual alliance of man, elf, dwarf, wizard and hobbit seek to destroy a powerful ring."
+   }
+ ];
+
+ const myDoc2 = [
+   {
+     "title": "The Fault in our stars",
+     "text": "Some infinities are bigger than others, a writer we used to like taught us that"
+   },
+   {
+     "title": "Of The Alchemist",
+     "text": "And, when you want something, all the universe conspires in helping you to achieve it."
+   }
+ ];
+
+ const wrongJson = [
+   {
+   "key": "Alice in Wonderland",
+   "value": "Alice falls into a rabbit hole and enters a world full of imagination."
+ }
+ ];
+
+ const invalidJson = "key Alice in Wonderland value Alice falls into a rabbit hole and enters a world full of imagination.";
+
+ const emptyJson = '';
+
+ let index;
+
+ describe('Inverted index', () => {
+   beforeEach(() => {
+     index = new Index();
+   });
+
+   describe('Check Input data', () => {
+     it('assert that the file is parsed correctly', () => {
+       expect(index.validJson('myDoc', JSON.stringify(myDoc))).toBe('Correct file');
+     });
+
+     it('tests for an empty file', () => {
+       expect(index.validJson('emptyJson', emptyJson)).toBe('Empty file');
+     });
+
+     it('assert that error is returned if the file is wrong', () => {
+       expect(index.validJson('wrongJson', JSON.stringify(wrongJson))).toBe('No title or text');
+     });
+
+     it('Checks for an invalid Json file', () => {
+       expect(index.validJson('invalidJson', invalidJson)).toBe('Invalid JSON file');
+     });
+   });
+
+   describe('Read book data', () => {
+     beforeEach(() => {
+       index.createIndex('myDoc', JSON.stringify(myDoc));
+     });
+
+     it('assert JSON file is not empty', () => {
```

Comment:
### Critical

Your assertion actually tests that an index exists, it does not assert that a JSON file is empty.